server/repository/assignments.go

```diff
@@ -0,0 +1,151 @@
+package repository
+
+import (
+	"database/sql"
+	"fmt"
+
+	"github.com/src-d/code-annotation/server/model"
+)
+
+// Assignments repository
+type Assignments struct {
+	db *sql.DB
+}
+
+// NewAssignments returns a new Assignments repository
+func NewAssignments(db *sql.DB) *Assignments {
+	return &Assignments{db: db}
+}
+
+// ErrNoAssignmentsInitialized is the error returned when the Assignments of a
+// User are requested for a given Experiment, but they have not been yet created
+var ErrNoAssignmentsInitialized = fmt.Errorf("No assignments initialized")
+
+// Initialize builds the assignments for the given user and experiment IDs
+func (repo *Assignments) Initialize(userID int, experimentID int) ([]*model.Assignment, error) {
+	tx, err := repo.db.Begin()
+	if err != nil {
+		return nil, err
+	}
+
+	insert, err := tx.Prepare("INSERT INTO assignments VALUES ($1, $2, $3, $4, $5)")
+	if err != nil {
+		return nil, fmt.Errorf("DB error: %v", err)
+	}
+
+	rows, err := repo.db.Query(
+		"SELECT id FROM file_pairs WHERE experiment_id=$1", experimentID)
+	if err != nil {
+		return nil, fmt.Errorf("Error getting file_pairs from the DB: %v", err)
+	}
+	defer rows.Close()
+
+	duration := 0
+
+	for rows.Next() {
+		var pairID int
+		rows.Scan(&pairID)
+
+		_, err := insert.Exec(userID, pairID, experimentID, nil, duration)
+		if err != nil {
+			return nil, fmt.Errorf("DB error: %v", err)
+		}
+	}
+
+	err = tx.Commit()
+	if err != nil {
+		return nil, fmt.Errorf("DB error: %v", err)
+	}
+
+	return repo.GetAll(userID, experimentID)
+}
+
+// Create stores an Assignment into the DB, and returns that new Assignment
+func (repo *Assignments) Create(as *model.Assignment) error {
+
+	_, err := repo.db.Exec(
+		`INSERT INTO assignments (user_id, pair_id, experiment_id, answer, duration)
+		VALUES ($1, $2, $3, $4, $5)`,
+		as.UserID, as.PairID, as.ExperimentID, as.Answer, as.Duration)
+
+	if err != nil {
+		return err
+	}
+
+	as, err = repo.Get(as.UserID, as.PairID)
+	return err
+}
+
+// getWithQuery builds a Assignment from the given sql QueryRow. If the
+// Assignment does not exist, it returns nil, nil
+func (repo *Assignments) getWithQuery(queryRow *sql.Row) (*model.Assignment, error) {
+	var as model.Assignment
+
+	err := queryRow.Scan(
+		&as.UserID, &as.PairID, &as.ExperimentID, &as.Answer, &as.Duration)
+
+	switch {
+	case err == sql.ErrNoRows:
+		return nil, nil
+	case err != nil:
+		return nil, fmt.Errorf("Error getting assignment from the DB: %v", err)
+	default:
+		return &as, nil
+	}
+}
+
+// Get returns the Assignment for the given user and pair IDs. If the Assignment
+// does not exist, it returns nil, nil
+func (repo *Assignments) Get(userID, pairID int) (*model.Assignment, error) {
+	return repo.getWithQuery(repo.db.QueryRow(
```

Comment:
This can return many assignments:
>_We considered that the same `FilePair` can be assigned to the same `User` many times in different experiments, so it can not be `PRIMARY KEY`_